File: solutions/elixir/line-up/2/lib/line_up.ex
defmodule LineUp do

  @message_start "you are the "
  @message_end " customer we serve today. Thank you!"

  defp suffix(n)
    when rem(n, 100) in [11, 12, 13],
    do: "th"

  defp suffix(n) do
    case rem(n, 10) do
      1 -> "st"
      2 -> "nd"
      3 -> "rd"
      _ -> "th"
    end
  end

  def format(name, number) do
    "#{name}, " <>
    "#{@message_start}#{number}#{suffix(number)}" <>
    "#{@message_end}"
  end

end

File: solutions/javascript/line-up/2/line-up.js
export const format = (name, number) => {
  const lastTwoDigits = number % 100;
  const lastDigit = number % 10;
  let ordinalSuffix = 'th'

  if (lastDigit === 1 && lastTwoDigits !== 11) {
    ordinalSuffix = 'st'
  } else if (lastDigit === 2 && lastTwoDigits !== 12) {
    ordinalSuffix = 'nd'
  } else if (lastDigit === 3 && lastTwoDigits !== 13) {
    ordinalSuffix = 'rd'
  }

  return `${name}, you are the ${number}${ordinalSuffix} customer we serve today. Thank you!`
}
